app/controllers/sessions_controller.rb

```diff
@@ -38,4 +38,33 @@ def validate_otp
       render nothing: true, status: :unauthorized
     end
   end
+
+  def validate_qrcode
+    signin_params = params
+      .permit(:account_id, :code)
+      .to_hash
+      .symbolize_keys
```

Comment:
We could use destructuring here since we won't use the signin_params, but its values.

```suggestion
    :account_id, :code = params
      .permit(:account_id, :code)
      .to_hash
      .symbolize_keys
      .values_at(:account_id, :code)
```
It simplifies how we'll use :code and :account_id. WDYT?

Reply:
That's true! Done here: 2f7fa38ae843a061dae32cb3634c374892fcb37a
It was also possible to remove `to_hash` and `symbolize_keys` calls.